src/python/pants/base/specs_parser_test.py

```diff
@@ -66,6 +72,7 @@ def assert_spec_parsed(build_root: Path, spec_str: str, expected_spec: Spec) ->
         (":root", address_literal("", "root")),
         ("//:root", address_literal("", "root")),
         ("a", dir_literal("a")),
+        ("a@k=v", address_literal("a", parameters={"k": "v"})),
```

Comment:
Feels like we might want to cover a few more cases? e.g., multiple parameters, and keys/values with more than one character? I know these are covered in address_test.py but it's good to verify that things work when parsed as specs.